File: web-sdk/src/domain/models/command/chat-command.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

// Client -> Channel commands
export const ChatCommands = [
  'updateCartProduct',
  'clearCart',
  'goToCart',
] as const;

export type ChatCommand = (typeof ChatCommands)[number];

// The callback can be synchronous or return a promise. The SDK waits for it
// to settle before marking the triggering action as done.
export type ChatCommandCallback = (payload: unknown) => void | Promise<void>;

File: flutter-sdk/lib/domain/models/command/chat_command.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

// Ids of the client -> channel commands the chat can trigger. Registering a
// callback for one of these ids runs it instead of the built-in remote call.
// These ids are reserved: any other id registered through registerCommand
// handles the matching channel -> client custom command request.
abstract final class ChatCommand {
  static const String updateCartProduct = 'updateCartProduct';
  static const String clearCart = 'clearCart';
  static const String goToCart = 'goToCart';
}

typedef ChatCommandCallback = void Function(dynamic payload);
